test/core.jl

```diff
@@ -23,57 +23,41 @@ end
     model = MLJFlux.NeuralNetworkRegressor()
     model.batch_size= 3
     @test MLJFlux.collate(model, X, y) ==
-        [(Xmatrix'[:,1:3], y[1:3]),
-         (Xmatrix'[:,4:6], y[4:6]),
-         (Xmatrix'[:,7:9], y[7:9]),
-         (Xmatrix'[:,10:10], y[10:10])]
+        ([Xmatrix'[:,1:3], Xmatrix'[:,4:6], Xmatrix'[:,7:9], Xmatrix'[:,10:10]], [y[1:3], y[4:6], y[7:9], y[10:10]])
 
     # NeuralNetworClassifier:
     y = categorical(['a', 'b', 'a', 'a', 'b', 'a', 'a', 'a', 'b', 'a'])
     model = MLJFlux.NeuralNetworkClassifier()
     model.batch_size = 3
     data = MLJFlux.collate(model, X, y)
-    @test first.(data) ==
-        [Xmatrix'[:,1:3], Xmatrix'[:,4:6],
-          Xmatrix'[:,7:9], Xmatrix'[:,10:10]]
-    @test last.(data) ==
-        [[1 0 1; 0 1 0], [1 0 1; 0 1 0],
-         [1 1 0; 0 0 1], reshape([1; 0], (2,1))]
+    print(data)
+    @test first.(data) == (Xmatrix'[:,1:3], [1 0 1; 0 1 0])
 
     # MultitargetNeuralNetworRegressor:
     ymatrix = rand(10, 2)
     y = MLJBase.table(ymatrix) # a rowaccess table
     model = MLJFlux.NeuralNetworkRegressor()
     model.batch_size= 3
-    @test MLJFlux.collate(model, X, y) ==
-        [(Xmatrix'[:,1:3], ymatrix'[:,1:3]),
-         (Xmatrix'[:,4:6], ymatrix'[:,4:6]),
-         (Xmatrix'[:,7:9], ymatrix'[:,7:9]),
-         (Xmatrix'[:,10:10], ymatrix'[:,10:10])]
+    @test MLJFlux.collate(model, X, y) == 
+        ([Xmatrix'[:,1:3], Xmatrix'[:,4:6], Xmatrix'[:,7:9], Xmatrix'[:,10:10]],
+                [ymatrix'[:,1:3], ymatrix'[:,4:6], ymatrix'[:,7:9], ymatrix'[:,10:10]])
+
     y = Tables.columntable(y) # try a columnaccess table
     @test MLJFlux.collate(model, X, y) ==
-        [(Xmatrix'[:,1:3], ymatrix'[:,1:3]),
-         (Xmatrix'[:,4:6], ymatrix'[:,4:6]),
-         (Xmatrix'[:,7:9], ymatrix'[:,7:9]),
-         (Xmatrix'[:,10:10], ymatrix'[:,10:10])]
+        ([Xmatrix'[:,1:3], Xmatrix'[:,4:6], Xmatrix'[:,7:9], Xmatrix'[:,10:10]], 
+            [ymatrix'[:,1:3], ymatrix'[:,4:6], ymatrix'[:,7:9], ymatrix'[:,10:10]])
 
     # ImageClassifier
     Xmatrix = coerce(rand(6, 6, 1, 10), GrayImage)
     y = categorical(['a', 'b', 'a', 'a', 'b', 'a', 'a', 'a', 'b', 'a'])
     model = MLJFlux.ImageClassifier(batch_size=2)
 
     data = MLJFlux.collate(model, Xmatrix, y)
-    @test  first.(data) ==
-        [Float32.(cat(Xmatrix[1], Xmatrix[2], dims=4)),
-        Float32.(cat(Xmatrix[3], Xmatrix[4], dims=4)),
-        Float32.(cat(Xmatrix[5], Xmatrix[6], dims=4)),
-        Float32.(cat(Xmatrix[7], Xmatrix[8], dims=4)),
-        Float32.(cat(Xmatrix[9], Xmatrix[10], dims=4)),
-        ]
+    @test  first.(data) == (Float32.(cat(Xmatrix[1], Xmatrix[2], dims=4)), [1 0;0 1])
```

Comment:
We've reduced test coverage here. Could we just test the whole `data` array (sucking in the target test that follows)?